### PR TITLE
Add more AzureClient e2e tests for SharedTree

### PR DESCRIPTION
**Pull request by noencke**
## Description

This adds a smattering of e2e tests for SharedTree to provide additional coverage of its main features and APIs. These tests are not meant to be completely exhaustive of every edge case (that is the job of the ST unit tests), but instead to be a "breadth first" sampling of SharedTree capabilities to improve upon the (currently very light) e2e test coverage.

This also refactors the setup code for the two existing SharedTree tests into a helper function.

~TODO: There is a known bug in node identifier encoding that will cause one of the new tests to fail. This PR must be rerun after the fix goes in.~

### CODE REVIEW

File: packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts [hunk 1]
```diff
@@ -72,35 +72,28 @@ for (const testOpts of testMatrix) {
...
 					"test-user-name-1",
 				);
+
 				containerId = getContainerIdFromPayloadResponse(containerResponse);
 				({ container: container1 } = await client.getContainer(containerId, schema, "2"));
 
-				treeData = (container1.initialObjects.tree1 as ITree).viewWith(
-					treeConfiguration, // This is defined in schema.ts
-				);
+				treeData = (container1.initialObjects.tree1 as ITree).viewWith(treeConfiguration);
```
Comment (CraigMacomber) [inline packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts:92]: container schema is suppose to have strong typing so we should not need that cast. Maybe move container1's definition inside the if, and let it infer the type?

File: packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts [hunk 2]
```diff
@@ -189,5 +173,178 @@ for (const testOpts of testMatrix) {
...
+						["Age", 32],
+						["Favorite Snack", "Potato Chips"],
+						["Awake", true],
+					];
+
+					view.initialize(
+						new User({
+							name: "Pardes",
+							nicknames: ["Alex", "Duder"],
+							data: new Map(tags),
```
Comment (CraigMacomber) [inline packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts:204]: We should actually tolerate an iterable of key value pairs here, so thjis should work:

```suggestion
							data: tags,
```

I wouldn't be surprised if that compiles and fails at runtime, I know we have some issues here, but we should probably test it.

File: packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts (hunk 1 above)
Comment (noencke) [reply packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts]: Good idea, I got rid of the underspecified `let` binding so we can have the stronger typing.

File: packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts (hunk 2 above)
Comment (noencke) [reply packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts:199]: Good idea. It does compile, but it also fails at runtime as you predicted. These "known issues" - are they tracked?

File: packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts [hunk 3]
```diff
@@ -113,6 +106,22 @@ for (const testOpts of testMatrix) {
...
 			}
 
+			return {
+				containerId,
+				treeData,
+			};
+		}
+
+		/**
+		 * Scenario: test when an Azure Client container is created,
```
[the remark this answers is not in the document]
Comment (noencke) [reply packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts]: Yea these comments don't seem particularly useful, they are mostly redundant with the test title anyway. I just deleted them and added a brief comment to the initialization function instead.

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**